ee/clickhouse/queries/event_query.py

```diff
@@ -172,7 +172,7 @@ def _get_props(self, filters: List[Property], allow_denormalized_props: bool = F
                 params.update(filter_params)
             elif prop.type == "element":
                 query, filter_params = filter_element({prop.key: prop.value}, prepend="{}_".format(idx))
-                final.append("AND {}".format(query[0]))
+                final.append(f" AND {query if len(query) > 0 else '1=2'}")
```

Comment:
Hmm, if we know apriori that some specific thing would return an empty list here, it probably makes more sense to validate this before the query, vs. fixing it like this inside the query?

This way, the backend doesn't error out, and for the "correct" fix, we ensure that the frontend sends the right things?

Reply:
Your point is correct. However these are completely valid payloads, but bad UX.

See these issues: https://github.com/PostHog/posthog/issues/5501 + https://github.com/PostHog/posthog/issues/5503

Reply:
Hmm, the empty string, or a null value is a valid payload: represented by `['']` and `[None]`. However, the `[]` payload is not correct?That should instead be an `is_not_set` or `is_set` operator instead of the `exact` operator.

And in the valid payloads, the query wouldn't be empty.

I did read through those two issues, and erroring out in the case of empty makes more sense to me? Since when we return 0 results, the user can't figure out what went wrong, vs. realising they input (thanks to bad UX) an invalid value?

Reply:
[] is a valid payload - see how it can be constructed in the linked issues and how other endpoints handle it.

Reply:
Discussed this face-to-face :)